fleetspeak/src/client/daemonservice/execution/execution.go

```diff
@@ -133,38 +131,42 @@ func New(daemonServiceName string, cfg *dspb.Config, sc service.Context) (*Execu
 	}
 	ret.Out = ret.channel.Out
 
-	if *dirForward != "" {
-		t := time.Now().UTC().UnixNano()
-		prefix := fmt.Sprintf("fleetspeak.%s.%d", daemonServiceName, t)
-
-		fo, err := os.Create(path.Join(*dirForward, prefix+".stdout.log"))
-		if err != nil {
-			return nil, fmt.Errorf("failed to open the stdout file for --dir_forward: %v", err)
-		}
-		ret.cmd.Stdout = fo
+	if cfg.StdParams != nil && cfg.StdParams.ServiceName == "" {
+		log.Errorf("std_params is set, but service_name is empty. Ignoring std_params: %v", cfg.StdParams)
+		cfg.StdParams = nil
```

Comment:
Hm, I am not sure whether modifying a function input is a good idea. Normally, config parameters are deemed to be constant (which is not possible to enforce in Go unfortunately). What I would do instead is to create a local variable `stdParams := cfg.StdParams` and set it to `nil` if necessary.

Reply:
This has become the typical implementation of default configuration values in this project. It is fairly safe given how the client instantiates the service configs. However I went ahead and cloned the proto first thing, and we can consider doing the same elsewhere.

Reply:
> This has become the typical implementation of default configuration values in this project. It is fairly safe given how the client instantiates the service configs.

Well, this is something you are aware of but some external contributor might not be. In general the less internal project assumptions about something are made (that are not explicitly enforced in code), the better.

This is of course just the purist inside me talking.